internal/sapsystem/sapsystem.go

```diff
@@ -485,6 +505,31 @@ func NewSAPInstance(w sapcontrol.WebService) (*SAPInstance, error) {
 	return sapInstance, nil
 }
 
+func detectType(sapControl *SAPControl) (int, error) {
+	sapLocalhost, err := sapControl.findProperty("SAPLOCALHOST")
+	if err != nil {
+		return Unknown, err
+	}
+
+	var instanceType int
+	for _, instance := range sapControl.Instances {
```

Comment:
As I understand here, we loop through the instances but instanceType will be always the last of instances that has sapLocalhost, because we don't break the loop, it's an expected behavior? Maybe if we are only interested in the "first one", or "at least one" of the instances is sapLocalhost we can return into the cases of the switch without looping.

Because if in the instances no one is the sapLocalhost instanceType will be the zero value of int so zero, maybe if this is expected we can initialize the variable not to int but to the appropriate "enum value" like "Application" "Database" etc..